http-server/src/server.rs

```diff
@@ -106,7 +106,49 @@ impl Builder {
 	}
 
 	/// Finalizes the configuration of the server.
-	pub fn build(self, addr: SocketAddr) -> Result<Server, Error> {
+	///
+	/// ```rust
+	/// #[tokio::main]
+	/// async fn main() {
+	///   let mut builder = jsonrpsee_http_server::HttpServerBuilder::default();
+	///   // try several addresses
+	///   let addrs: &[std::net::SocketAddr] = &[
+	///       "127.0.0.1:10".parse::<std::net::SocketAddr>().unwrap(),
+	///       "127.0.0.1:0".parse::<std::net::SocketAddr>().unwrap(),
+	///   ];
+	///   let server = builder.build(addrs).unwrap();
+	/// }
+	/// ```
+	pub fn build(self, addrs: impl ToSocketAddrs) -> Result<Server, Error> {
```

Comment:
NOTE: this is `std::net::ToSocketAddrs` because `tokio::std::ToSocketAddrs` is sealed i.e, doesn't provide `to_socket_addrs` because we configure the Socket ourselves with some settings.